src/main/webapp/app/components/geneHistoryTooltip/gene-history-tooltip-utils.tsx

```diff
@@ -152,7 +154,7 @@ export function getTooltipHistoryList(history: FlattenedHistory[]) {
 
       tooltipHistoryList.set(locationIdentifier, [...prevRecords, record]);
     } else {
-      tooltipHistoryList.get(locationIdentifier).push(record);
+      tooltipHistoryList.get(locationIdentifier)?.push(record);
```

Comment:
No, but TypeScript can't figure that out. If you want to avoid this then I would need to set a locally variable. Do you want me to do that?

Reply:
It's ok not to. But is it because this is a map? Typically typescript is able to figure it out if the value is an object for instance.

Reply:
Yeah, you're calling a method which can return `undefined` so typescript can't guarantee that it won't return `undefined`. It will never happen, but the only way typescript can know for sure is if you use a local variable.

Reply:
Could use `!` in this situation so we don't pollute everything downstream with `undefined` checks

Reply:
@zhx828  what do you think?